src/main/java/com/drinchev/projectlabel/resources/ui/ProjectLabelBackgroundImage.java

```diff
@@ -0,0 +1,116 @@
+
+
+@Service(Level.PROJECT)
+public class ProjectLabelBackgroundImage {
+
+    private final static Logger LOG = Logger.getInstance(ProjectLabelStatusBarWidget.class);
+
+    private BufferedImage bufferedImage;
+
+    private String resultingImage;
+
+    private final Project project;
+
+    private final PreferencesReader preferences;
+
+    public ProjectLabelBackgroundImage(@NotNull Project project) {
+        this.project = requireNonNull(project);
+        this.preferences = new PreferencesReader(project, ProjectPreferences.getInstance(project), ApplicationPreferences.getInstance());
+    }
+
+    public void showImage() {
+        if (!shouldShowBackgroundImage()) {
+            hideImage();
+            return;
+        }
+        createImage();
+        setImageToIDE();
+    }
+
+    private boolean shouldShowBackgroundImage() {
+        return preferences.backgroundImagePosition() != BackgroundImagePosition.HIDDEN;
+    }
+
+    private void setImageToIDE() {
+        PropertiesComponent prop = projectLevelPropertiesComponent();
+        String opacity = String.valueOf(preferences.backgroundImageOpacity());
+        String imageProp = String.format("%s,%s,plain,%s", resultingImage, opacity, preferences.backgroundImagePosition().name().toLowerCase());
+        String prev_editor = prop.getValue(IdeBackgroundUtil.EDITOR_PROP);
+        prop.setValue(IdeBackgroundUtil.EDITOR_PROP, imageProp); // statt plain: tile (wiederholen), scale (zoomen)
+        String prev_frame = prop.getValue(IdeBackgroundUtil.FRAME_PROP);
+        prop.setValue(IdeBackgroundUtil.FRAME_PROP, imageProp);
+        LOG.warn("Previous editor image: " + prev_editor);
+        LOG.warn("Previous frame image: " + prev_frame);
+    }
+
+    private PropertiesComponent projectLevelPropertiesComponent() {
+        PropertiesComponent prop = project.getService(PropertiesComponent.class);
+        return prop;
+    }
+
+    public void updateImage() {
+        LOG.info("Updating project label background image.");
+        bufferedImage = null;
+        resultingImage = null;
+        showImage();
+    }
+
+    public void hideImage() {
+        PropertiesComponent prop = projectLevelPropertiesComponent();
+        if (!prop.getValue(IdeBackgroundUtil.EDITOR_PROP, "").contains("project-label") && !prop.getValue(IdeBackgroundUtil.FRAME_PROP, "").contains("project-label")) {
+            LOG.info("Not touching background image since not a project label image.");
+            return;
+        }
+        LOG.info("Hiding project label background image.");
+        prop.setValue(IdeBackgroundUtil.EDITOR_PROP, null);
+        prop.setValue(IdeBackgroundUtil.FRAME_PROP, null);
+    }
+
+
+    private void createImage() {
+        if (bufferedImage == null) {
+            try {
+                ProjectLabelAWTRenderer renderer = new ProjectLabelAWTRenderer(preferences);
+                Dimension preferredImageRation = renderer.getPreferredImageRatio();
```

Comment:
I guess this makes the label a bit spooky.

So it will never have the same `height` but it will rather depend on the length of the label : 

<img width="2560" alt="A" src="https://user-images.githubusercontent.com/202919/231186627-f89e668c-d801-48f9-8eb6-1a6279cb3924.png">
<img width="2560" alt="A1" src="https://user-images.githubusercontent.com/202919/231186632-bc353ea6-a671-41cc-97e9-fca9447d160b.png">
<img width="2560" alt="A1B" src="https://user-images.githubusercontent.com/202919/231186637-a767b4c7-82c9-4337-9e8c-13c856ef929a.png">

Do you think it will be better if we convert this to something like "20%" of the height of the screen ( so it also looks nice on a laptop size )?

Reply:
Yes, I'm also not entirely happy with the current solution.

I see the following options:

1.) **relative to the screen size**
This would mean we need to adapt the image whenever the screen size changes. (Display is plugged in/removed, Window changes Display)

2.) **relative to the window size**
Again, we would need to adapt it when the window changes size.

3.) **relative to the status bar widget font size**

4.) **Let the user select a font size in pixels**
Similar to the status bar widget, we let the user select a font size for the background image directly.

In all cases, it might be a good idea to somehow cap the maximum width of the image?

What's your take?

Reply:
Yeah this gets quite ugly with two windows on the same machine.

![image](https://user-images.githubusercontent.com/202919/231483760-64469091-aac3-4088-8b5b-17bc94fe8e73.png)

I guess I would prefer to have **this relative to the window size**. 

I'm not sure it would be cool if the user chooses the font-size in pixels as I'm also moving sometimes from my desktop screen to my laptop screen and in one case the label is just too big. It doesn't make sense to change the font-size each time I move from desktop to laptop screen.

In the Status Bar we have the benefit that is always fixed height no matter which screen you are looking at it. If you think we can make the background label way-smaller ( something like 100px fixed height ) it would be okay for all screens I guess.

Reply:
I've adapted the code to resize the label dynamically based on the window size. I kind of like it this way, give it a try and LMK what you think.

Reply:
Looks way nicer. 

![image](https://user-images.githubusercontent.com/202919/231715284-001e145b-18a7-4f72-9fc0-0cfadaf4b1e4.png)

The only "bug" I see is the bottom margin is smaller than the right margin

Reply:
Nice, I'll look into the margin issue and get back to you.